resize-loop.c

```diff
@@ -0,0 +1,78 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "client.h"
+
+#define MIN 2
+#define MAX 512
+#define SPEED 10
+static int size = MIN;
+
+static struct wleird_toplevel toplevel = {0};
+static const struct wl_callback_listener callback_listener;
+
+
+static void request_frame_callback(void) {
+	struct wl_callback *callback = wl_surface_frame(toplevel.surface.wl_surface);
+	wl_callback_add_listener(callback, &callback_listener, NULL);
+	wl_surface_commit(toplevel.surface.wl_surface);
+}
+
+static void callback_handle_done(void *data, struct wl_callback *callback,
+		uint32_t time_ms) {
+	if (callback != NULL) {
+		wl_callback_destroy(callback);
+	}
+
+	toplevel.surface.width = abs(size);
+	toplevel.surface.height = abs(size);
+	surface_render(&toplevel.surface);
+
+	size += SPEED;
+	if (size < 0 && size > -MIN) {
+		size = MIN;
+	} else if (size > 0 && size > MAX) {
+		size = -MAX;
```

Comment:
Note to self: we use negative values to avoid a jump from MAX to MIN.

Reply:
Yeah okay, maybe I should've added a comment about using the sign bit to indicate direction of the movement. Or just used a bool. :P